src/components/Task.jsx

```diff
@@ -1,52 +1,74 @@
 import {
   Box,
+  Button,
   CardContent,
   Checkbox,
   IconButton,
+  TextField,
   Typography,
 } from "@mui/material";
 import DeleteIcon from "@mui/icons-material/Delete";
 import ModeEditIcon from "@mui/icons-material/ModeEdit";
 import MoreVertIcon from "@mui/icons-material/MoreVert";
 import { useState } from "react";
+import { useCallback } from "react";
 
-function Task({ title }) {
+function Task({ todo, onChangeTask }) {
   const [done, setDone] = useState(false);
+  const [isEdit, setIsEdit] = useState(false);
 
+  const setEditStatus = useCallback(() => {
+    setIsEdit(!isEdit);
+  }, [isEdit]);
+  const onChangeTextBox = useCallback((e) => {
+    onChangeTask({ ...todo, title: e.target.value });
```

Comment:
The name of the event should be `onTaskChanged`.
And 
`onChangeTextBox` => `handleInputChanged`